pkg/reconciler/common/prune.go

```diff
@@ -240,30 +240,32 @@ func prunableNamespaces(ctx context.Context, k kubernetes.Interface, defaultPrun
 	return prunableNs, nil
 }
 
-func (pruner *Pruner) createAllJobContainers(nsConfig map[string]*pruneConfigPerNS) []corev1.Container {
-	var containers []corev1.Container
+func generateAllPruneCommands(nsConfig map[string]*pruneConfigPerNS) string {
+	var cmds string
 	for ns, con := range nsConfig {
-		jobContainers := pruner.createJobContainers(con, ns)
-		containers = append(containers, jobContainers...)
+		cmd := generatePruneCommand(con, ns)
+		cmds = cmds + " " + cmd
 	}
-	return containers
+
+	return cmds
 }
 
-func (pruner *Pruner) createJobContainers(nsConfig *pruneConfigPerNS, ns string) []corev1.Container {
-	var containers []corev1.Container
+func generatePruneCommand(nsConfig *pruneConfigPerNS, ns string) string {
+	cmd := pruneCommand(nsConfig, ns)
+	return cmd
+}
 
-	cmdArgs := pruneCommand(nsConfig, ns)
+func getJobContainer(cmdArgs, ns string, tknImage string) []corev1.Container {
+	cmd := "function prune() {   n=$1; a=$2; resources=$3; old_ifs=\" \"; IFS=\",\"; for r in $resources; do tkn $r delete -n=$n $a -f; done; IFS=$old_ifs; }; echo $conf; for c in $*; do   ns=$(echo $c | cut -d \";\" -f 1); args=$(echo $c | cut -d \";\" -f 2); resources=$(echo $c | cut -d \";\" -f 3); prune $ns $args $resources; done;"
 	containerName := SimpleNameGenerator.RestrictLengthWithRandomSuffix("pruner-tkn-" + ns)
 	container := corev1.Container{
 		Name:                     containerName,
-		Image:                    pruner.tknImage,
+		Image:                    tknImage,
 		Command:                  []string{"/bin/sh", "-c"},
-		Args:                     []string{cmdArgs},
+		Args:                     []string{cmd, "-s", cmdArgs},
```

Comment:
let us make it
```golang
Command: []string{"/bin/sh", "-c", cmd}
Args:     []string{"-s", cmdArgs}
```

Reply:
I started with that, but earlier was getting file name too long, and `createContainerError`.
But now not getting that error. Updating it.